backend/clickhouse/trace_row.go

```diff
@@ -0,0 +1,163 @@
+package clickhouse
+
+import (
+	"fmt"
+	"time"
+)
+
+type TraceRow struct {
+	ProjectId          uint32
+	SecureSessionId    string
+	Timestamp          time.Time
+	TraceId            string
+	SpanId             string
+	ParentSpanId       string
+	TraceState         string
+	SpanName           string
+	SpanKind           string
+	ServiceName        string
+	ResourceAttributes map[string]string
+	SpanAttributes     map[string]string
+	Duration           int64
+	StatusCode         string
+	StatusMessage      string
+	Events             []Event
+	Links              []Link
+}
+
+type Event struct {
+	Timestamp  time.Time
+	Name       string
+	Attributes map[string]string
+}
+
+type Link struct {
+	TraceId    string
+	SpanId     string
+	TraceState string
+	Attributes map[string]string
+}
+
+func NewTraceRow(timestamp time.Time) *TraceRow {
+	traceRow := &TraceRow{
+		Timestamp:   timestamp,
+		ServiceName: "",
```

Comment:
Omit this. In Clickhouse, the default value for string columns is already an empty string ([docs](https://clickhouse.com/docs/en/sql-reference/statements/create/table#default_values)) (very similar to Go zero values).